ksql-functional-tests/src/test/resources/query-validation-tests/cast.json

```diff
@@ -61,6 +61,88 @@
       "outputs": [
         {"topic": "OUTPUT", "value": ",,,,"}
       ]
+    },
+    {
+      "name": "decimal to decimal",
+      "statements": [
+        "CREATE STREAM TEST (foo DECIMAL(4,2)) WITH (kafka_topic='test_topic', value_format='DELIMITED');",
+        "CREATE STREAM OUT AS SELECT cast(foo AS DECIMAL(5,3)) FROM TEST;"
+      ],
+      "inputs": [
+        {"topic":  "test_topic", "value": "10.12"},
+        {"topic":  "test_topic", "value": "01.00"},
+        {"topic":  "test_topic", "value": "00.00"}
+      ],
+      "outputs": [
+        {"topic":  "OUT", "value": "10.120"},
+        {"topic":  "OUT", "value": "01.000"},
+        {"topic":  "OUT", "value": "00.000"}
+      ]
+    },
+    {
+      "name": "integer to decimal",
+      "statements": [
+        "CREATE STREAM TEST (foo INTEGER) WITH (kafka_topic='test_topic', value_format='AVRO');",
+        "CREATE STREAM OUT AS SELECT cast(foo AS DECIMAL(2,1)) as VAL FROM TEST;"
+      ],
+      "inputs": [
+        {"topic":  "test_topic", "value": {"foo": 1}},
+        {"topic":  "test_topic", "value": {"foo": 0}},
+        {"topic":  "test_topic", "value": {"foo": -1}},
+        {"topic":  "test_topic", "value": {"foo": 10}}
+      ],
+      "outputs": [
+        {"topic":  "OUT", "value": {"VAL": "1.0"}},
+        {"topic":  "OUT", "value": {"VAL": "0.0"}},
+        {"topic":  "OUT", "value": {"VAL": "-1.0"}},
+        {"topic":  "OUT", "value": {"VAL": null}}
+      ]
+    },
+    {
+      "name": "double to decimal",
+      "statements": [
+        "CREATE STREAM TEST (foo DOUBLE) WITH (kafka_topic='test_topic', value_format='AVRO');",
+        "CREATE STREAM OUT AS SELECT cast(foo AS DECIMAL(2,1)) as VAL FROM TEST;"
+      ],
+      "inputs": [
+        {"topic":  "test_topic", "value": {"foo": 0.1}},
+        {"topic":  "test_topic", "value": {"foo": 1.1}},
+        {"topic":  "test_topic", "value": {"foo": 0.0}},
+        {"topic":  "test_topic", "value": {"foo": 0.99}},
+        {"topic":  "test_topic", "value": {"foo": 0.10}},
+        {"topic":  "test_topic", "value": {"foo": 0.01}}
+      ],
+      "outputs": [
+        {"topic":  "OUT", "value": {"VAL": "0.1"}},
+        {"topic":  "OUT", "value": {"VAL": "1.1"}},
+        {"topic":  "OUT", "value": {"VAL": "0.0"}},
+        {"topic":  "OUT", "value": {"VAL": "1.0"}},
+        {"topic":  "OUT", "value": {"VAL": "0.1"}},
+        {"topic":  "OUT", "value": {"VAL": "0.0"}}
+      ]
+    },
+    {
+      "name": "string to decimal",
```

Comment:
Consider adding negative tests for casting numbers too large, (+'ve and -'ve), as these test what error the user sees. Same for cast to invalid type.

Reply:
unfortunately errors in casting end up just being swallowed and `null` values are returned, so it's not especially useful as a QTT. improving the user experience around bad values is out of scope unfortunately

Reply:
Good point, and agreed!

Reply:
I might actually be wrong about that - I just noticed that `cast.json` already has some exceptional cases. I guess maybe only `KsqlFunctionalException` gets raised? Now the question is whether in the case of a bad cast it's better to return `null` as with most KSQL errors or to fail hard...